src/amo/utils/index.js

```diff
@@ -48,3 +50,34 @@ export const getCanonicalURL = ({
 |}): string => {
   return `${_config.get('baseURL')}${locationPathname}`;
 };
+
+type QueryParams = { [key: string]: any };
+
+type MakeInternalExternalURLParams = {|
+  _config: typeof config,
+  urlString: string,
+  internalQueryParams?: QueryParams,
+  externalQueryParams?: QueryParams,
+|};
+
+export const makeInternalExternalURL = ({
+  _config = config,
+  externalQueryParams = {},
+  internalQueryParams = {},
+  urlString,
+}: MakeInternalExternalURLParams) => {
+  const baseURL = _config.get('baseURL');
+  const urlParts = url.parse(urlString, true);
+
+  const queryParams =
+    !urlParts.protocol || baseURL.includes(urlParts.host)
```

Comment:
Did you consider just using this logic as a function like `isInternalUrl(url)`? I feel like that might make everything simpler (maybe the tests, too) but I guess it's fine as is. It was just a thought.